[CONTP-1569] Add KSM labels_as_tags / annotations_as_tags / tags to CRD (#2937)
DataDog/datadog-operator: docs/configuration.v2alpha1.md

| features.helmCheck.collectEvents | CollectEvents set to `true` enables event collection in the Helm check (Requires Agent 7.36.0+ and Cluster Agent 1.20.0+) Default: false |
| features.helmCheck.enabled | Enables the Helm check. Default: false |
| features.helmCheck.valuesAsTags | ValuesAsTags collects Helm values from a release and uses them as tags (Requires Agent and Cluster Agent 7.40.0+). Default: {} |

> | features.kubeStateMetricsCore.annotationsAsTags | AnnotationsAsTags maps Kubernetes annotations to Datadog tags, scoped to the KSM check. Outer key is the Kubernetes resource kind; inner map is annotation name -> Datadog tag name. Example:   annotationsAsTags:     pod:       tags_datadoghq_com_version: version Annotation names must match the transformation done by kube-state-metrics (e.g. `tags.datadoghq.com/version` becomes `tags_datadoghq_com_version`). Note: the top-level `global.kubernetesResourcesAnnotationsAsTags` configures this at the agent level via environment variables. AnnotationsAsTags here writes into the KSM check instance config and applies only to KSM metrics. |

Reviewer: ```suggestion
| features.kubeStateMetricsCore.annotationsAsTags | AnnotationsAsTags maps Kubernetes annotations to Datadog tags, scoped to the KSM check. The outer key is the Kubernetes resource kind, and the inner map is annotation name -> Datadog tag name. Example:   annotationsAsTags:     pod:       tags_datadoghq_com_version: version <br>Annotation names must match the transformation done by kube-state-metrics (for example, `tags.datadoghq.com/version` becomes `tags_datadoghq_com_version`). <br>**Note**: The top-level `global.kubernetesResourcesAnnotationsAsTags` configures this at the Agent level via environment variables. AnnotationsAsTags here writes to the KSM check instance config and applies only to KSM metrics. |
```

| features.kubeStateMetricsCore.collectCrMetrics | `CollectCrMetrics` defines custom resources for the kube-state-metrics core check to collect.  The datadog agent uses the same logic as upstream `kube-state-metrics`. So is its configuration. The exact structure and existing fields of each item in this list can be found in: https://github.com/kubernetes/kube-state-metrics/blob/main/docs/metrics/extend/customresourcestate-metrics.md |
| features.kubeStateMetricsCore.conf.configData | ConfigData corresponds to the configuration file content. |
| features.kubeStateMetricsCore.conf.configMap.items | Maps a ConfigMap data `key` to a file `path` mount. |
| features.kubeStateMetricsCore.conf.configMap.name | Is the name of the ConfigMap. |
| features.kubeStateMetricsCore.enabled | Enables Kube State Metrics Core. Default: true |

> | features.kubeStateMetricsCore.labelsAsTags | LabelsAsTags maps Kubernetes labels to Datadog tags, scoped to the KSM check. Outer key is the Kubernetes resource kind (e.g. "pod", "node", "deployment"); inner map is label name -> Datadog tag name. Example:   labelsAsTags:     pod:       app: app     node:       zone: zone Note: the top-level `global.kubernetesResourcesLabelsAsTags` configures this at the agent level via environment variables. LabelsAsTags here writes into the KSM check instance config and applies only to KSM metrics. |

Reviewer: ```suggestion
| features.kubeStateMetricsCore.labelsAsTags | LabelsAsTags maps Kubernetes labels to Datadog tags, scoped to the KSM check. The outer key is the Kubernetes resource kind (for examp,e, "pod", "node", "deployment"), and the inner map is label name -> Datadog tag name. Example:   labelsAsTags:     pod:       app: app     node:       zone: zone <br>**Note**: The top-level `global.kubernetesResourcesLabelsAsTags` configures this at the Agent level via environment variables. LabelsAsTags here writes to the KSM check instance config and applies only to KSM metrics. |
```
